packages/larva/scripts/config/gulpfile.js

```diff
@@ -46,46 +47,82 @@ const stylelint = ( file ) => {
 	gulp.src( file ).pipe( gulpStylelint( stylelintOpts ) );
 };
 
-// We need an empty stream for the else condition when 
-// minifying CSS below. Reference: https://stackoverflow.com/a/30000562
-const emptyStream = () => {
-	var pass = through2.obj();
+/**
+ * PostCSS plugin that does nothing, for conditionally minifying CSS.
+ *
+ * @param {Object} style  PostCSS Root object for current CSS.
+ * @param {Object} result PostCSS Result object containing transformed CSS.
+ * @return {Result} PostCSS Result object.
+ */
+const postCssNoop = ( style, result ) => {
+	result.root = postCss.parse( style );
+	return result;
+};
+
+/**
+ * Add `!important` declarations to all rules, for use with older themes that
+ * have high specificity.
+ *
+ * Adapted from the `postcss-importantly` npm package, updated to work with
+ * latest version of PostCSS.
+ *
+ * @param {Object} style  PostCSS Root object for current CSS.
+ * @param {Object} result PostCSS Result object containing transformed CSS.
+ * @return {Result} PostCSS Result object.
+ */
+const declareImportanceForAll = ( style, result ) => {
+	const root = postCss.parse( style );
+
+	root.walkRules( ( rule ) => {
+		if ( 'atrule' === rule.parent.type && 'media' !== rule.parent.name ) {
+			return;
+		}
 
-	process.nextTick( pass.end.bind(pass) );
-	return pass;
-}
+		return rule.each( ( decl ) => {
+			if ( ! decl.value || decl.important ) {
+				return;
+			}
+
+			decl.value += '!important';
+
+			return decl;
+		} );
+	} );
+
+	result.root = root;
+
+	return result;
+};
 
 /**
  * Build CSS
- * 
+ *
  * Used for both prod and dev commands.
- * 
+ *
  * Consider updating the name 'minify' to 'post' if/when
  * more Post CSS functionality is added.
- * 
- * @param {function} done 
+ *
+ * @param {Function} done  Post-build callback.
  * @param {boolean} minify Run post CSS and minify output.
  */
 const buildScss = ( done, minify = false ) => {
+	const postCssPlugins = [];
 
-	// This is redundant, but was having issue with conditionally
-	// minifying within the same stream with 
-	// .pipe( minify ? postcss( [ cssnano() ] ) : emptyStream() )
-	// In interest of time, use separate streams for now.
-
-	if ( true === minify ) {
-		gulp.src( './entries/*.scss' )
-			.pipe( gulpStylelint( stylelintOpts ) )
-			.pipe( sass( sassOpts ).on( 'error', sass.logError ) )
-			.pipe( postcss( [ cssnano() ] ) )
-			.pipe( gulp.dest( cssDest ) );
+	if ( minify ) {
+		postCssPlugins.push( cssnano() );
 	} else {
-		gulp.src( './entries/*.scss' )
-			.pipe( gulpStylelint( stylelintOpts ) )
-			.pipe( sass( sassOpts ).on( 'error', sass.logError ) )
-			.pipe( gulp.dest( cssDest ) );
+		postCssPlugins.push( postCssNoop );
 	}
 
+	gulp.src( './entries/*.scss' )
+		.pipe( gulpStylelint( stylelintOpts ) )
+		.pipe( sass( sassOpts ).on( 'error', sass.logError ) )
+		.pipe( gulpPostCss( postCssPlugins ) )
+		.pipe( gulp.dest( cssDest ) )
+		.pipe( gulpPostCss( [ declareImportanceForAll ] ) )
```

Comment:
This gulpfile is used in themes to build the local CSS as well, so they will also generate an `XX-important` stylesheet for each entry even when its not needed – do you think we can have a flag or separate gulp script so this only happens when required? There are some utilities that exist to get CLI args that can be added to the prod-scss.js script.